pkg/apis/componentconfig/types.go

```diff
@@ -0,0 +1,32 @@
+/*
```

Comment:
The `componentconfig` API in k8s is an API for the configuration options different system components like the API server, kubelet, etc, accept.  @krancour was right to model it in this case, but I expect it will change as we refactor to the kubernetes controller frameworks.